pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -29,18 +29,25 @@ static __always_inline void postgres_batch_enqueue_wrapper(conn_tuple_t *tuple,
     postgres_batch_enqueue(event);
 }
 
-// Reads a message header from the given context. Returns true if the header was read successfully, false otherwise.
-static __always_inline bool read_message_header(pktbuf_t pkt, struct pg_message_header* header) {
+// Reads a message header from the given context.
+// Returns the result of the read: successfully read header, end of packet reached, or fragmented message detected.
+static __always_inline postgres_read_result_t read_message_header(pktbuf_t pkt, struct pg_message_header* header) {
     u32 data_off = pktbuf_data_offset(pkt);
     u32 data_end = pktbuf_data_end(pkt);
-    // Ensuring that the header is in the buffer.
-    if (data_off + sizeof(struct pg_message_header) > data_end) {
-        return false;
-    }
-    pktbuf_load_bytes(pkt, data_off, header, sizeof(struct pg_message_header));
-    // Converting the header to host byte order.
-    header->message_len = bpf_ntohl(header->message_len);
-    return true;
+
+    if (data_off + sizeof(struct pg_message_header) <= data_end) {
+        // header fits within the buffer boundaries.
+        pktbuf_load_bytes(pkt, data_off, header, sizeof(struct pg_message_header));
+        // Converting the header to host byte order.
+        header->message_len = bpf_ntohl(header->message_len);
+
+        return READ_OK;
+    }
+    if (data_off == data_end) {
+        return READ_END;
+    }
+    // the offset is outside the packet boundaries
+    return READ_FRAGMENTED;
 }
```

Comment:
Reporting `READ_OK`, `READ_END` and `READ_FRAGMENTED` does not bring any value to the PR.
Eventually, you're only checking if the result was `READ_OK` or not which is equivalent for the previous version of a boolean

Reply:
I increase counter `pg_msg_counts->fragmented_packets` when `READ_FRAGMENTED` and stop iterating tail calls when `READ_END`

Reply:
you're using it only once, and we can get the same outcome without modifying that internal function.

Reply:
I've restored previous `read_message_header()` function